Mako stub crash when too much data are pushed in

<!--

Classify what kind of issue this is:

/kind bug

-->

## Expected Behaviour
It outputs all data

## Actual Behaviour
It doesn't output anything, the container exits with status code 0 and `q.Store()` hangs returning any error

## Steps to Reproduce the Problem

1. Try to run the broker tests in eventing repo with increased parallelism and longer pace configuration, like `1000:30,2000:30,1000:30,2000:30`
2. Follow both the log of aggregator and mako stub container
3. When the aggregator will start to publish data to store, the mako-stub will quit with status code 0

## Additional Info
You can try with PR https://github.com/knative/eventing/pull/1917 too, since it doubles the generated data

More info:
```
panic: close of closed channel

goroutine 6 [running]:
main.(*server).ShutdownMicroservice(0xc000088068, 0x9d64e0, 0xc00015a300, 0xc00000c2e0, 0xc000088068, 0xc00015a300, 0xc00007fba8)
	/home/francesco/go/src/knative.dev/pkg/test/mako/stub-sidecar/main.go:99 +0x2e
knative.dev/pkg/third_party/mako/proto/quickstore_go_proto._Quickstore_ShutdownMicroservice_Handler(0x89c700, 0xc000088068, 0x9d64e0, 0xc00015a300, 0xc000170190, 0x0, 0x9d64e0, 0xc00015a300, 0x0, 0x0)
	/home/francesco/go/src/knative.dev/pkg/third_party/mako/proto/quickstore_go_proto/quickstore.pb.go:347 +0x23e
knative.dev/pkg/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc00008c180, 0x9da5a0, 0xc00008c780, 0xc000158100, 0xc00010e270, 0xd56d58, 0x0, 0x0, 0x0)
	/home/francesco/go/src/knative.dev/pkg/vendor/google.golang.org/grpc/server.go:998 +0x470
knative.dev/pkg/vendor/google.golang.org/grpc.(*Server).handleStream(0xc00008c180, 0x9da5a0, 0xc00008c780, 0xc000158100, 0x0)
	/home/francesco/go/src/knative.dev/pkg/vendor/google.golang.org/grpc/server.go:1278 +0xda6
knative.dev/pkg/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00007c6d0, 0xc00008c180, 0x9da5a0, 0xc00008c780, 0xc000158100)
	/home/francesco/go/src/knative.dev/pkg/vendor/google.golang.org/grpc/server.go:717 +0x9f
created by knative.dev/pkg/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/home/francesco/go/src/knative.dev/pkg/vendor/google.golang.org/grpc/server.go:715 +0xa1
```

--- a/test/mako/stub-sidecar/main.go
+++ b/test/mako/stub-sidecar/main.go
@@ -24,6 +24,7 @@ import (
 	"net"
 	"os"
 	"strings"
+	"sync"
 
 	"github.com/golang/protobuf/jsonpb"
 
@@ -36,7 +37,8 @@ const (
 )
 
 type server struct {
-	stopCh chan struct{}
+	stopOnce sync.Once
+	stopCh   chan struct{}
 }
 
 func (s *server) Store(ctx context.Context, in *qspb.StoreInput) (*qspb.StoreOutput, error) {
@@ -96,7 +98,7 @@ func makeRow(prototype []string, points map[string]string) []string {
 }
 
 func (s *server) ShutdownMicroservice(ctx context.Context, in *qspb.ShutdownInput) (*qspb.ShutdownOutput, error) {
-	close(s.stopCh)
+	s.stopOnce.Do(func() { close(s.stopCh) })
 	return &qspb.ShutdownOutput{}, nil
 }
 

